pkgs/development/compilers/zig/default.nix

```diff
@@ -1,25 +1,47 @@
 { stdenv, fetchFromGitHub, cmake, llvmPackages, libxml2, zlib }:
 
-stdenv.mkDerivation rec {
-  version = "0.5.0";
+llvmPackages.stdenv.mkDerivation rec {
+  version = "0.6.0";
   pname = "zig";
 
   src = fetchFromGitHub {
     owner = "ziglang";
     repo = pname;
     rev = version;
-    sha256 = "0xyl0riakh6kwb3yvxihb451kqs4ai4q0aygqygnlb2rlr1dn1zb";
+    sha256 = "13dwm2zpscn4n0p5x8ggs9n7mwmq9cgip383i3qqphg7m3pkls8z";
   };
 
   nativeBuildInputs = [ cmake ];
-  buildInputs = [ llvmPackages.clang-unwrapped llvmPackages.llvm libxml2 zlib ];
+  buildInputs = [
+    llvmPackages.clang-unwrapped
+    llvmPackages.llvm
+    llvmPackages.lld
+    libxml2
+    zlib
+  ];
+
+  patches = [ ./llvm10_polly.patch ];
+
+  postPatch = ''
+    export llvm_extras=-Wl,${llvmPackages.llvm}/lib/LLVMPolly.so
+    substituteAllInPlace CMakeLists.txt
+  '';
```

Comment:
I think something like this should work (will also need to add `substituteAll` to the args):

```nix
  patches = [
    (substituteAll {
        src = ./llvm10_polly.patch;
        llvm_extras = "-Wl,${llvmPackages.llvm}/lib/LLVMPolly.so";
    })
  ];
```